include/swift/AST/DiagnosticsSema.def

```diff
@@ -1184,9 +1184,9 @@ WARNING(conditional_downcast_coercion,none,
       (Type, Type))
 
 WARNING(literal_conditional_downcast_to_coercion,none,
-        "conditional downcast from literal to %0 always fails; "
+        "conditional downcast from literal of inferred type %0 to unrelated type %1 always fails; "
```

Comment:
Small nitpick, through out all diagnostics we consistently use "inferred as"(we can find a few examples if search for "inferred as" filtering *.def files)
```suggestion
        "conditional downcast from literal inferred as %0 to unrelated type %1 always fails; "
```
So maybe we could go with that here as well? @AnthonyLatsis WDYT?

Reply:
Yeah, either one is fine to me as well =]

Reply:
Please wrap this to 80 chars too.